src/Assignment3/tests/Assignment3/search/SearchEx3Test.kt

```diff
@@ -0,0 +1,54 @@
+package Assignment3.search
+
+import org.junit.jupiter.api.Test
+
+import org.junit.jupiter.api.Assertions.*
+import java.lang.ArithmeticException
+import kotlin.math.abs
+
+internal class SearchEx3Test {
+
+    val search = SearchEx3()
+
+    @Test
+    fun testPow1() {
+        assert(search.pow(2.0, 10) == 1024.0)
+    }
+
+    @Test
+    fun testPow2() {
+        assert(search.pow(2.0, -2) == 0.25)
+    }
+
+    @Test
+    fun testPow3() {
+        assert(abs(search.pow(0.2, 3) - 0.008) <= 0.000001)
```

Comment:
Maybe try this `Assert.assertEquals(expected, actual, precision)` for compareson between double/float?